cmd/cluster/support/delete.go

```diff
@@ -0,0 +1,176 @@
+	}
+
+	return deleteCmd
+}
+
+func newDeleteOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *deleteOptions {
+
+	return &deleteOptions{
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *deleteOptions) complete(cmd *cobra.Command, args []string) error {
+
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "Provide exactly one internal cluster ID")
+	}
+
+	o.clusterID = args[0]
+	o.output = o.GlobalOptions.Output
+
+	return nil
+}
+
+func (o *deleteOptions) run() error {
+
+	ctx := context.Background()
+
+	// Create an OCM client to talk to the cluster API
+	token := os.Getenv("OCM_TOKEN")
+	if token == "" {
+		ocmToken, err := ctlutil.GetOCMAccessToken()
+		if err != nil {
+			log.Fatalf("OCM token not set. Please configure by using the OCM_TOKEN environment variable or the ocm cli")
+			os.Exit(1)
+		}
+		token = *ocmToken
+	}
+	connection, err := sdk.NewConnectionBuilder().
+		Tokens(token).
+		Build()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't build connection: %v\n", err)
+		os.Exit(1)
+	}
+	defer connection.Close()
+
+	// Stop here if dry-run
+	if isDryRun {
+		return nil
+	}
+
+	// confirmSend prompt to confirm
+	confirmSend()
+
+	// Get cluster resource
+	clusterResource := connection.ClustersMgmt().V1().Clusters().Cluster(o.clusterID)
+	clusterResponse, err := clusterResource.Get().SendContext(ctx)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't retrieve cluster: %v\n", err)
+		os.Exit(1)
+	}
+	cluster := clusterResponse.Body()
+
+	deleteRequest, err := createDeleteRequest(connection, cluster, o.limitedSupportReasonID)
+	if err != nil {
+		fmt.Printf("failed post call %q\n", err)
+	}
+	deleteResponse, err := sendDeleteRequest(deleteRequest)
+	if err != nil {
+		fmt.Printf("Failed to get delete call response: %q\n", err)
+	}
+
+	err = checkDelete(deleteResponse)
+	if err != nil {
+		fmt.Printf("check for delete call failed: %q", err)
+	}
+
+	return nil
+}
+
+// createDeleteRequest sets the delete API and returns a request
+func createDeleteRequest(ocmClient *sdk.Connection, cluster *v1.Cluster, reasonID string) (request *sdk.Request, err error) {
+
+	targetAPIPath := "/api/clusters_mgmt/v1/clusters/" + cluster.ID() + "/limited_support_reasons/" + reasonID
+
+	request = ocmClient.Delete()
+	err = arguments.ApplyPathArg(request, targetAPIPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse API path '%s': %v", targetAPIPath, err)
+	}
+	return request, nil
+}
+
+func sendDeleteRequest(request *sdk.Request) (*sdk.Response, error) {
+
+	response, err := request.Send()
+	if err != nil {
+		return nil, fmt.Errorf("cannot send request: %q", err)
+	}
+	return response, nil
+}
+
+// checkDelete checks the response from delete API call
+// 204 if success, otherwise error
+func checkDelete(response *sdk.Response) error {
+
+	var badReply *support.BadReply
+	body := response.Bytes()
+	if response.Status() == 204 {
```

Comment:
You can use `http.StatusNoContent` here :)

Reply:
Makes sense. Thanks!